pedpy/data/geometry.py

```diff
@@ -66,3 +66,211 @@ def add_obstacle(self, obstacle: Polygon) -> None:
                 f"The obstacle {obstacle} is not inside the walkable area of "
                 "the geometry and thus will be ignored!"
             )
+
+
+###############################################################################
+# Measurement Area
+###############################################################################
+class MeasurementArea:
+    """Areas to study pedestrian dynamics.
+
+    A measurement area is defined as an area, which is convex, simple, and
+    covers a non-zero area.
+    """
+
+    _polygon: shapely.Polygon
+    _frozen = False
+
+    def __init__(self, coordinates: Any):
+        """Create a measurement area from the given input.
+
+        The measurement area may be a convex, simple area which covers a
+        non-zero area.
+
+        Args:
+            coordinates: A sequence of (x, y [,z]) numeric coordinate pairs, or
+                an array-like with shape (N, 2). Also, can be a sequence of
+                shapely.Point objects. Passing a wkt representation of a polygon
+                is also allowed.
+        """
+        try:
+            if isinstance(coordinates, shapely.Polygon):
+                self._polygon = coordinates
+            elif isinstance(coordinates, str):
+                self._polygon = shapely.from_wkt(coordinates)
```

Comment:
I think you need to do more here because this can contain all sorts of geometry types, polygon, multipolygon, lines etc...

IIRC from_wkt can be relatively nicely used with a GeometryColletion and then you can inspect this and check for the correct types.